sparkbox/sparkle
# spkl-29: True Test Setup

Description:
1. True Test Setup:
- Each file with mixins, utilities, & functions have corresponding `_test.scss` file.
- All test files are being pulled into `test/test.scss`
2.  Unit Test Docs

To Test:
Run `npm test`

If you'd like to run a sample test, 
1. Create a folder with a `_mixins.scss` and `_test.scss` file. 
2. Add this code  
```
/_test.scss
@import 'true';
@import './mixin';

$map: (
  'size': (
    'sml': 10px
  )
);

@include describe('map-get()') {
    @include it('should return the value from a map request') {
      @include assert-equal(
        map-get($map, 'size', 'sml'), 10px
      );
    }
  }
```

```
/_mixin.scss
@function map-get($map, $keys...) {
    @each $key in $keys {
      $map: map-get($map, $key);
    }
    @return $map;
  }
```
3. Import the `_test.scss` file into `test/test.scss`
